### PR TITLE
Convert MemberDeviceInfo to ES6 class

### PR DESCRIPTION
**Pull request by richvdh**
... kinda just because.

### CODE REVIEW

File: src/components/views/rooms/MemberDeviceInfo.js [hunk 1]
```diff
@@ -14,41 +14,43 @@ See the License for the specific language governing permissions and
...
-module.exports = React.createClass({
-    displayName: 'MemberDeviceInfo',
-    propTypes: {
-        userId: React.PropTypes.string.isRequired,
-        device: React.PropTypes.object.isRequired,
-    },
+export default class MemberDeviceInfo extends React.Component {
+    constructor(props) {
+        super(props);
+        this.onVerifyClick = this.onVerifyClick.bind(this);
```
Comment (ara4n) [inline src/components/views/rooms/MemberDeviceInfo.js]: i'm surprised that these are needed?

Comment (ara4n) [inline src/components/views/rooms/MemberDeviceInfo.js]: oh, right, https://facebook.github.io/react/docs/reusable-components.html#no-autobinding.

**Comment by ara4n** [comment]
this looks okay to me, although I'm slightly dubious as to whether we want to be proactively converting to ES6 classes; what actually are the benefits?  It feels like we lose a bunch of `function` keywords, but instead have to put important stuff like PropTypes at the bottom of the file, and do all the manual `this` binding.  Plus it makes the codebase increasingly inconsistent stylistically.

I'm not going to veto it, but just wondering just how aggressively we want to go in this dir.

**Comment by richvdh** [comment]
so yeah, let's not proactively convert them. We have, however, decided that we are going to use ES6 classes in future (or at least, that's what Dave and I have been doing) so stylistic arguments against them have been somewhat lost already.

The story behind this particular change is: I was going to make some bigger changes; looked at it and thought "meh it's trivial, might as well switch to an ES6 class first", then decided against the bigger changes. So we end up with just the ES6 change.

**Comment by ara4n** [comment]
ok. would be interested to know what the rationale was for opting in future for ES6, but that can be taken elsewhere :)

**Comment by ara4n** [comment]
lgtm

**Pull request merged**